openwebstart/src/main/java/com/openwebstart/jvm/ui/dialogs/ConfigurationDialog.java

```diff
@@ -40,26 +53,31 @@ public ConfigurationDialog() {
         updateStrategyComboBox.setSelectedItem(RuntimeManagerConfig.getStrategy());
 
         final JLabel defaultVendorLabel = new JLabel(translator.translate("dialog.jvmManagerConfig.vendor.text"));
-        final String[] combinedList = JavaRuntimeManager.getAllVendors(RuntimeManagerConfig.getDefaultRemoteEndpoint());
-        final JComboBox<String> vendorComboBox = new JComboBox<>(combinedList);
+        vendorComboBox = new JComboBox();
+        backgroundExecutor.execute(() -> updateVendorComboBox(RuntimeManagerConfig.getDefaultRemoteEndpoint()));
         vendorComboBox.setEditable(true);
-        vendorComboBox.setSelectedItem(RuntimeManagerConfig.getVendor());
+
 
         final JLabel defaultUpdateServerLabel = new JLabel(translator.translate("dialog.jvmManagerConfig.defaultServerUrl.text"));
         final JTextField defaultUpdateServerField = new JTextField();
+        originalBackground = defaultUpdateServerField.getBackground();
         defaultUpdateServerField.setText(Optional.ofNullable(RuntimeManagerConfig.getDefaultRemoteEndpoint()).map(URL::toString).orElse(""));
+        defaultUpdateServerField.addFocusListener(new MyFocusAdapter());
 
         final JCheckBox allowAnyUpdateServerCheckBox = new JCheckBox(translator.translate("dialog.jvmManagerConfig.allowServerInJnlp.text"));
         allowAnyUpdateServerCheckBox.setSelected(RuntimeManagerConfig.isNonDefaultServerAllowed());
 
-
         final JLabel supportedVersionRangeLabel = new JLabel(translator.translate("dialog.jvmManagerConfig.versionRange.text"));
         final JTextField supportedVersionRangeField = new JTextField();
         supportedVersionRangeField.setText(Optional.ofNullable(RuntimeManagerConfig.getSupportedVersionRange()).map(VersionString::toString).orElse(""));
 
         final JButton okButton = new JButton(translator.translate("action.ok"));
         okButton.addActionListener(e -> {
             try {
+                if (defaultUpdateServerField.getBackground() == ERROR_BACKGROUND) {
```

Comment:
Objects.equals instead of ==
Next to this the background color is maybe not the best validator to check if everything is fine.